### PR TITLE
Refactored all add() calls from SortedSet to only pass single values

### PR DESCRIPTION
**Pull request by dangrie158**
Before this fix, the index passed all tokens via an apply call, pushing all parameters on the stack.

This behaviour causes a RangeError when more tokens are added than the JS implementation can handle (Firefox 44 -> 262143, Node.js -> 65535, evaluated using this test: http://stackoverflow.com/questions/22747068/is-there-a-max-number-of-arguments-javascript-functions-can-accept)
this fix changes the code so that now only a single value gets passed per call. This way no change to the current API is neccesary.

The code was refactored in a way that it now passes only a single argument at a time, avoiding the usage of apply.

The only parts of the code that still use apply are the EventEmitters and the Plugin subsystem which should be non critical for this type of error.

### CODE REVIEW

File: lib/sorted_set.js [hunk 1]
```diff
@@ -224,7 +224,9 @@ lunr.SortedSet.prototype.union = function (otherSet) {
 
   unionSet = longSet.clone()
 
-  unionSet.add.apply(unionSet, shortSet.toArray())
+  for(var i = 0, shortSetElements = shortSet.toArray(); i < shortSetElements.length; i++){
+    unionSet.add(shortSetElements[i]);
```
Comment (d0ugal) [inline lib/sorted_set.js]: tiny nitpick - none of the other code includes explicitly semicolons. We should probably try and match that style.

**Comment by d0ugal** [comment]
One tiny comment, otherwise the change looks good.

**Comment by dangrie158** [comment]
Thanks I just fixed the line.
I found it really hard to follow the semicolon less style because its almost in the muscle memory to press it when your not used to it.

Maybe I can add a .jshintrc at some point with the [lastsemic](http://jshint.com/docs/options/#lastsemic) option set at some point? That would at least supress all warnings for a automatic linter like I have in Atom.

**Comment by d0ugal** [comment]
> I found it really hard to follow the semicolon less style because its almost in the muscle memory to press it when your not used to it.

I agree, I prefer JavaScript with semicolons.

Anyway, this change looks good to me now! :+1:

**Comment by olivernn** [comment]
This is merged in the [latest release](https://github.com/olivernn/lunr.js/releases/tag/v0.7.0) now, thanks a lot the fix.

I actually think adding some kind of linter to the build would be a fantastic idea, it certainly helps any would be contributors to get their patch applied quicker (sadly I'm probably still the bottleneck though!) and would help ensure a decent level of consistency in the code base.

Also semicolon free style is clearly better... :trollface: :fire:
